src/library-authoring/component-info/ComponentUsage.tsx

```diff
@@ -0,0 +1,94 @@
+import { FormattedMessage } from '@edx/frontend-platform/i18n';
+import { Collapsible } from '@openedx/paragon';
+import { Link } from 'react-router-dom';
+
+import AlertError from '../../generic/alert-error';
+import Loading from '../../generic/Loading';
+import { useFetchIndexDocuments, type ContentHit } from '../../search-manager';
+import { useComponentDownstreamLinks } from '../data/apiHooks';
+import messages from './messages';
+
+interface ComponentUsageProps {
+  usageKey: string;
+}
+
+type ComponentUsageTree = Record<string, {
+  key: string,
+  contextName: string,
+  links: ContentHit['breadcrumbs'][1][],
+}>;
+
+export const ComponentUsage = ({ usageKey }: ComponentUsageProps) => {
+  const {
+    data: dataDownstreamLinks,
+    isError: isErrorDownstreamLinks,
+    error: errorDownstreamLinks,
+    isLoading: isLoadingDownstreamLinks,
+  } = useComponentDownstreamLinks(usageKey);
+
+  const downstreamKeys = dataDownstreamLinks || [];
+
+  const {
+    data: downstreamHits,
+    isError: isErrorIndexDocuments,
+    error: errorIndexDocuments,
+    isLoading: isLoadingIndexDocuments,
+  } = useFetchIndexDocuments({
+    filter: [`usage_key IN ["${downstreamKeys.join('","')}"]`],
+    limit: downstreamKeys.length,
+    attributesToRetrieve: ['usage_key', 'breadcrumbs', 'context_key'],
+    enabled: !!downstreamKeys.length,
+  });
+
+  if (isErrorDownstreamLinks || isErrorIndexDocuments) {
+    return <AlertError error={errorDownstreamLinks || errorIndexDocuments} />;
+  }
+
+  if (isLoadingDownstreamLinks || isLoadingIndexDocuments) {
+    return <Loading />;
+  }
```

Comment:
Selecting a component that is not used in any course shows loading component indefinately as `isLoadingIndexDocuments` stays true when the api is disabled.

```suggestion
  if (isLoadingDownstreamLinks || (isLoadingIndexDocuments && !!downstreamKeys.length)) {
    return <Loading />;
  }
```